tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -125,69 +148,127 @@ def _install_agent_on_node(self, agent_package: Path) -> None:
+            log.info("Collecting logs on test machine [%s]...", self.context.node.name)
+            self.execute_script_on_node(self.context.test_source_directory/"orchestrator"/"scripts"/"collect-logs", sudo=True)
 
             # Copy the tarball to the local logs directory
-            remote_path = self._context.node_home_directory/"logs.tgz"
-            local_path = Path.home()/'logs'/'vm-logs-{0}.tgz'.format(self._node.name)
-            self._log.info(f"Copying {self._node.name}:{remote_path} to {local_path}")
-            self._node.shell.copy_back(remote_path, local_path)
-        except Exception as e:
-            self._log.warning(f"Failed to collect logs from the test machine: {e}")
+            remote_path = self.context.remote_working_directory / "logs.tgz"
+            local_path = Path.home()/'logs'/'vm-logs-{0}.tgz'.format(self.context.node.name)
+            log.info("Copying %s:%s to %s", self.context.node.name, remote_path, local_path)
+            self.context.node.shell.copy_back(remote_path, local_path)
+        except:  # pylint: disable=bare-except
+            log.exception("Failed to collect logs from the test machine")
 
-    def execute(self, scenario: Callable[[Context], None]) -> None:
+    def execute(self, node: Node, test_suite: List[Type[AgentTest]]) -> None:
         """
-        Executes the given scenario
+        Executes each of the AgentTests in the given List. Note that 'test_suite' is a list of test classes, rather than
+        instances of the test class (this method will instantiate each of these test classes).
         """
+        self._set_context(node)
+
+        log.info("")
+        log.info("**************************************** [Setup] ****************************************")
+        log.info("")
+
+        failed: List[str] = []
+
         try:
             self._setup()
+
             try:
                 self._setup_node()
-                scenario(self._context)
+
+                log.info("")
+                log.info("**************************************** [%s] ****************************************", self._metadata.full_name)
+                log.info("")
+
+                results: List[str] = []
+
+                for test in test_suite:
+                    try:
+                        log.info("******************** [%s]", test.__name__)
+                        log.info("")
+
+                        test(self.context).run()
+
+                        result = f"[Passed] {test.__name__}"
+
+                        log.info("")
+                        log.info("******************** %s", result)
+                        log.info("")
+
+                        results.append(result)
+                    except:  # pylint: disable=bare-except
+                        result = f"[Failed] {test.__name__}"
+
+                        log.info("")
+                        log.exception("******************** %s\n", result)
```

Comment:
can we log error msg too. I think test name is not enough for debugging.

Reply:
log.exception logs the exception message and traceback